test/pamela/HTN/symbol-ref-htn.main.htn.edn

```diff
@@ -0,0 +1,372 @@
+  :argsmap
+  {"condition"
+   {:args
+    [{:type :field-ref, :names [field3]}
+     {:type :method-arg-ref, :names [method-arg]}],
+    :type :equal}},
+  :display-args [(= field3 method-arg)],
+  :display-name "Tell",
+  :edges [:hedge-66],
+  :incidence-set #{:hedge-65},
+  :name "tell",
+  :plant-id "bsm1",
+  :type :htn-primitive-task,
+  :uid :hpt-39},
+ :hpt-40
+ {:args
+  [{:type :equal,
+    :args
+    [{:type :field-ref, :names [field2 field1-1]}
+     {:type :method-arg-ref, :names [method-arg]}]}],
+  :argsmap
+  {"condition"
+   {:args
+    [{:type :field-ref, :names [field2 field1-1]}
+     {:type :method-arg-ref, :names [method-arg]}],
+    :type :equal}},
+  :display-args [(= field2.field1-1 method-arg)],
+  :display-name "Tell",
+  :edges [:hedge-67],
+  :incidence-set #{:hedge-66},
+  :name "tell",
+  :plant-id "bsm1",
+  :type :htn-primitive-task,
+  :uid :hpt-40},
+ :hpt-41
+ {:args
+  [{:type :equal,
+    :args
+    [{:type :method-arg-ref, :names [pclass-instance-arg field1-1]}
+     {:type :method-arg-ref, :names [method-arg]}]}],
+  :argsmap
+  {"condition"
+   {:args
+    [{:type :method-arg-ref, :names [pclass-instance-arg field1-1]}
+     {:type :method-arg-ref, :names [method-arg]}],
+    :type :equal}},
+  :display-args [(= pclass-instance-arg.field1-1 method-arg)],
+  :display-name "Tell",
+  :edges [:hedge-68],
+  :incidence-set #{:hedge-67},
+  :name "tell",
+  :plant-id "bsm1",
+  :type :htn-primitive-task,
+  :uid :hpt-41},
+ :hpt-42
+ {:args [42],
+  :argsmap {"arg" 42},
+  :display-args [42],
+  :display-name "Simple Method",
+  :edges [:hedge-69],
+  :incidence-set #{:hedge-68},
+  :name "simple-method",
+  :type :htn-primitive-task,
+  :uid :hpt-42},
+ :hpt-43
+ {:args [{:type :state-variable, :name state-var}],
+  :argsmap {"arg" {:name state-var, :type :state-variable}},
+  :display-args [state-var],
+  :display-name "Simple Method",
+  :edges [:hedge-70],
+  :incidence-set #{:hedge-69},
+  :name "simple-method",
+  :type :htn-primitive-task,
+  :uid :hpt-43},
+ :hpt-44
+ {:args [field3],
+  :argsmap {"arg" field3},
+  :display-args [field3],
+  :display-name "Simple Method",
+  :edges [:hedge-71],
+  :incidence-set #{:hedge-70},
+  :name "simple-method",
+  :type :htn-primitive-task,
+  :uid :hpt-44},
+ :hpt-45
+ {:args
+  [{:type :pclass-ctor,
+    :uid :pclass-3,
+    :pclass pclass1,
+    :args [43],
+    :fields
+    {field1-1 {:access :private, :observable false, :initial 43}},
+    :ancestry [[:pclass-1 field2]]}],
+  :argsmap
+  {"arg"
+   {:ancestry [[:pclass-1 field2]],
+    :args [43],
+    :fields
+    {field1-1 {:access :private, :initial 43, :observable false}},
+    :pclass pclass1,
+    :type :pclass-ctor,
+    :uid :pclass-3}},
+  :display-args [field2.field1-1],
+  :display-name "Simple Method",
+  :edges [:hedge-72],
+  :incidence-set #{:hedge-71},
+  :name "simple-method",
+  :type :htn-primitive-task,
+  :uid :hpt-45},
+ :hpt-46
+ {:args [42],
+  :argsmap {"arg" 42},
```

Comment:
**This is admittedly a very hairy example!**

The original Pamela was `(simple-method pclass-instance-arg.field1-1)`
Since `field1` is passed as a pclass instance with an initial value of field1-1, I don't think it is correct to inline that initial value (42) here, since it might change.  Instead, I think we need to come up with a representation analogous to whatever we end up using for cases like `:hpt-45`.